launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -328,4 +328,17 @@ static String findJarsDir(String sparkHome, String scalaVersion, boolean failIfN
     return libdir.getAbsolutePath();
   }
 
+  /**
+   * Add "m" as the default suffix unit when no explicit unit is given.
+   */
+  static String addDefaultMSuffixIfNeeded(String memoryString) {
+    if (memoryString.chars().allMatch(Character::isDigit)) {
+      System.err.println("Memory setting without explicit unit (" +
+        memoryString + ") is taken to be in MB by default! For details check SPARK-32293.");
```

Comment:
Given that we are documenting 'm' is the suffix we use if not specified, do we need this message to stderr ?